src/PresentationalComponents/SystemsTable/SystemsTable.js

```diff
@@ -1,66 +1,93 @@
-    ]);
-    const [rows, setRows] = useState([]);
-    const [sortBy, setSortBy] = useState({});
+import { systemReducer } from '../../AppReducer';
+
+const SystemsTable = ({
+    systemsFetchStatus,
+    fetchSystems,
+    systems,
+    intl,
+    filters,
+    setFilters,
+    history,
+    selectedTags
+}) => {
+    const inventory = useRef(null);
+    const [InventoryTable, setInventory] = useState();
+    const store = useStore();
     const results = systems.meta ? systems.meta.count : 0;
     const [searchText, setSearchText] = useState(filters.display_name || '');
     const debouncedSearchText = debounce(searchText, 800);
     const [filterBuilding, setFilterBuilding] = useState(true);
-
     const sortIndices = {
         0: 'display_name',
         1: 'hits',
         2: 'last_seen'
     };
 
-    const onSort = (_event, index, direction) => {
+    const onSort = ({ index, direction }) => {
         const orderParam = `${direction === 'asc' ? '' : '-'}${sortIndices[index]}`;
-        setSortBy({ index, direction });
         setFilters({ ...filters, sort: orderParam, offset: 0 });
     };
 
-    const onSetPage = (pageNumber) => {
-        const newOffset = pageNumber * filters.limit - filters.limit;
-        setFilters({ ...filters, offset: newOffset });
-    };
+    useEffect(() => {
+        (async () => {
+            const {
+                inventoryConnector,
+                mergeWithEntities,
+                INVENTORY_ACTION_TYPES
+            } = await insights.loadInventory({
+                react: React,
+                reactRouterDom,
+                pfReactTable
+            });
+            getRegistry().register({
+                ...mergeWithEntities(
+                    systemReducer(
+                        [
+                            { title: intl.formatMessage(messages.name), transforms: [sortable], key: 'display_name' },
+                            { title: intl.formatMessage(messages.numberRuleHits), transforms: [sortable, cellWidth(15)], key: 'hits' },
+                            { title: intl.formatMessage(messages.lastSeen), transforms: [sortable], key: 'updated' }
+                        ],
+                        INVENTORY_ACTION_TYPES
+                    )
+                )
+            });
+
+            const { InventoryTable } = inventoryConnector(store);
+            setInventory(() => InventoryTable);
+        })();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
     useEffect(() => {
         filters.display_name === undefined ? setSearchText('') : setSearchText(filters.display_name);
     }, [filters.display_name]);
 
     useEffect(() => {
-        const filter = { ...filters };
-        const text = searchText.length ? { display_name: searchText } : {};
-        delete filter.display_name;
-        setFilters({ ...filter, ...text });
+        if (Boolean(searchText) !== Boolean(filters.display_name)) {
```

Comment:
for posterity, this breaks url update, cuz we're not removing the search field from the filter object, (that's what builds the url)